helsingborg-stad/helsingborg-io-sls-api
# 🚛 Move timeslots to booking api

## Explain the changes you’ve made

I have moved `timeslots-api` to `booking-api`, and converted all the JS files in booking-api to TS.

## Explain why these changes are made

Timeslots API only had one lambda, which was related to the booking service. It also had a helper function that was needed in booking API. Therefore it made sense to combine the two APIs.

## Explain your solution

After moving the files and updating references, I renamed the files from `.js` to `.ts` and fixed the typing issues. Since some types were needed in several files, I moved all the types into a separate file `helpers/types.ts`. 

## How to test

Concrete example:

1. Checkout this branch
2. Run unit tests
3. Run `sls deploy` within service folder for `booking-api`
4. Verify that the `booking/getTimeSlots` endpoint works as intended

## Tested environments

- [] Your local Serverless environment.

## NOTE: Remember to merge https://github.com/helsingborg-stad/modularity-mhbooking/pull/3 after this is merged

---

## Checklist for reviewers

Here are a few points that are useful for reviewers to check.

The short list of the (maybe) most important points are:

- Does this PR work as expected?
- Does the PR properly only concern a single feature/bug/thing?
- Is the code clear and concise?
- Will the changes impact existing functionality?
- Are there any linting errors (ESLint) that should be resolved?
- Do the automatic GitHub checks pass?

However it is good if a reviewer can be as thorough as possible,
within reasonable limits of course.

### Functionality

- Does this PR work as expected?
- If this is a new feature, does it add value, or is it a sign of feature-creep?
- Will the changes impact existing functionality?
- Does the changes create inconsistencies somewhere else?
- Does the code work on different platforms if applicable?
- How does the code handle if unexpected values are passed to functions etc.?

### Readability, syntax, formatting, style

- Is the code clear and concise?
- Has styling rules (Prettier) been applied to included files?
- Are functions/variables/etc. aptly named?

### Design

- Is the code properly planned and designed?
- Is code organized/separated in a logical fashion?
- Does the code follow clean code practices to a reasonable degree?
- Is the code self-documenting enough?
- Is the code in sync with existing patterns/technologies?
- Does the code reuse existing functionality where applicable?
- Is the code designed with reusability in mind?
- Does the code have hard-coded values that can be refactored?

### Patterns, idioms, tests, best practices

- Are there any linting errors (ESLint) that should be resolved?
- Is the code properly typed (TypeScript)?
- Does the code utilize language features to their full potential?

### Debugging, testing, configuring

- Is appropriate logging used?
- Have tests been added/modified to accommodate the changes?
- Is the feature sufficiently configurable in case tweaks are required in the future?

### Performance, reliability, scalability

- Is the code well optimized?
- Does the code scale?
- Does the code handle failure scenarios?

### Security

- Is there any obvious potential for exploits and security threats?
- Does code check proper authorization where applicable?

### Etiquettes

- Are commits atomic (does only one thing per commit)?
- Does the PR properly only concern a single feature/bug/thing?
- Are commit messages well-written?
- Do the automatic GitHub checks pass?
